### PR TITLE
Proposed 0.18.0

### PR DESCRIPTION
**Pull request by intelliot**
Big change:
- Convert to TypeScript

Small changes:
- Use ES Module syntax (Babel still compiles these to common.js modules for distribution)
- Docs: Improve escrow creation example
- Fix type errors
- Fix lint errors

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -11,6 +11,14 @@
 + Fix lint errors ([#786](https://github.com/ripple/ripple-lib/pull/786), [#808](https://github.com/ripple/ripple-lib/pull/808))
 + [Update ripple-keypairs dependency to 0.10.1](https://github.com/ripple/ripple-lib/pull/805)
 
+The SHA checksums for the browser version of this release can be found below.
```
Comment (wilsonianb) [inline HISTORY.md]: Can we use sha256 instead?

**Comment by intelliot** [comment]
I think we can gradually take care of the TODO/Fix comments later on. Here are PRs for a couple of them: https://github.com/ripple/ripple-lib/pull/832, https://github.com/ripple/ripple-lib/pull/833

File: HISTORY.md (hunk 1 above)
Comment (intelliot) [reply HISTORY.md]: Added in ae2ccd98873c4a1d09c2fb95dbca25e9eca28a37

File: src/ledger/parse/payment-channel.ts [hunk 2]
```diff
@@ -1,23 +1,49 @@
...
 import {parseTimestamp} from './utils'
 import {removeUndefined, dropsToXrp} from '../../common'
 
+
+export type PaymentChannel = {
+  Sequence: number,
+  Account: string,
+  Amount: string,
+  Balance: string,
+  PublicKey: number,
```
[the remark this answers is not in the document]
Comment (intelliot) [reply src/ledger/parse/payment-channel.ts]: Fixed in e8ac4f3d

File: test/fixtures/responses/prepare-escrow-creation.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "txJSON": "{\"Flags\":2147483648,\"TransactionType\":\"EscrowCreate\",\"Account\":\"r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59\",\"Destination\":\"rpZc4mVfWUif9CRoHRKKcmhu1nx2xktxBo\",\"Amount\":\"10000\",\"CancelAfter\":464908910,\"LastLedgerSequence\":8820051,\"Fee\":\"12\",\"Sequence\":23}",
+  "txJSON": "{\"Flags\":2147483648,\"TransactionType\":\"EscrowCreate\",\"Account\":\"r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59\",\"Destination\":\"rpZc4mVfWUif9CRoHRKKcmhu1nx2xktxBo\",\"Amount\":\"10000\",\"CancelAfter\":536544000,\"FinishAfter\":464908910,\"LastLedgerSequence\":8820051,\"Fee\":\"12\",\"Sequence\":23}",
```
[the remark this answers is not in the document]
Comment (intelliot) [reply test/fixtures/responses/prepare-escrow-creation.json]: Fixed in b7d636d

File: src/server/server.ts [hunk 4]
```diff
@@ -23,12 +21,12 @@ function getServerInfo(): Promise<GetServerInfoResponse> {
   return common.serverInfo.getServerInfo(this.connection)
 }
 
-function getFee(): Promise<number> {
+function getFee(): Promise<string> {
```
[the remark this answers is not in the document]
Comment (intelliot) [reply src/server/server.ts]: Fixed in 6d39e07

File: .babelrc [hunk 5]
```diff
@@ -1,4 +1,3 @@
 {
-  "presets": ["es2015", "stage-1"],
-  "plugins": ["syntax-flow", "transform-flow-strip-types"]
+  "presets": ["es2015", "stage-1", "flow"]
```
Comment (tpkiddle) [inline .babelrc]: Is the "flow" preset required?

Comment (intelliot) [reply .babelrc]: Good catch, https://github.com/ripple/ripple-lib/pull/834 will remove it

**Pull request merged**